basictracer/span.py

```diff
@@ -0,0 +1,109 @@
+from __future__ import absolute_import
+
+import time
+import re
+
+from opentracing import Span
+from .context import Context
+from .util import generate_id
+
+class BasicSpan(Span):
+    """BasicSpan extends opentracing.Span
+    """
+
+    def __init__(self, tracer, raw):
+        """ Starts and returns a new basictracer span.
+
+        Extends the opentracing Span class.
+        """
+        super(BasicSpan, self).__init__(tracer)
+        self.raw = raw
+
+    def set_operation_name(self, operation_name):
+        self.raw.operation_name = operation_name
+        return super(BasicSpan, self).set_operation_name(operation_name)
+
+    def set_tag(self, key, value):
+        if self.raw.tags is None:
+            self.raw.tags = {}
+        self.raw.tags[key] = value
+        return super(BasicSpan, self).set_tag(key, value)
+
+    def log_event(self, event, payload=None):
+        self.raw.logs.append(LogData(event=event, payload=payload))
+        return super(BasicSpan, self).log_event(event, payload)
+
+    def log(self, **kwargs):
```

Comment:
hmm, I didn't realize we did kwargs, I hate those.

@bensigelman why not do proper named params? I remember your point about param ordering, but to me having kwargs is a bigger evil, in terms of possibility of typos etc.